index.js

```diff
@@ -350,7 +358,11 @@ async function createHttps(ssl, log) {
     }
 
     log('https | listening on %d', port);
-    server = https.createServer(finalHttpsOptions, ssl.handler);
+    if(h2) {
+      server = require('http2').createSecureServer(finalHttpsOptions, ssl.handler)
+    } else {
+      server = https.createServer(finalHttpsOptions, ssl.handler);
```

Comment:
Should this now also 'lazily' require `https`? If both lines are consistent you could 

```suggestion
      server = require(h2 ? 'http2' : 'https').createServer(finalHttpsOptions, ssl.handler);
```

Reply:
We could late http2 uses `.createSecureServer`, but still, yes we could lazy require `https` (and `http` for that matter) which would clean up the variable shadowing that is occurring. — Speaking of which, this repo could benefit from eslint. I may do a follow-up PR with that.